File: tests/expected/rect.nim

type
  Rectangle = object
    height: int
    length: int
proc is_square(self: Rectangle): bool =
  return self.height == self.length


proc show() =
  var r = Rectangle(height: 1, length: 1)
  assert(r.is_square())
  r = Rectangle(height: 1, length: 2)
  assert(not (r.is_square()))
  let h = r.height
  let l = r.length
  echo h
  echo l

proc main() =
  show()

main()

File: tests/expected/rect.go
package main

import (
	"fmt"
)

// This file implements a rectangle class

type Rectangle struct {
	height int
	length int
}

func (self Rectangle) is_square() bool {
	return self.height == self.length
}

func Show() {
	var r Rectangle = Rectangle{height: 1, length: 1}
	if !(r.is_square()) {
		panic("assert")
	}
	r = Rectangle{height: 1, length: 2}
	if !(!(r.is_square())) {
		panic("assert")
	}
	fmt.Printf("%v\n", r.height)
	fmt.Printf("%v\n", r.length)
}

func main() {
	Show()
}
